components/map/basic/src/index.js

```diff
@@ -0,0 +1,248 @@
+import React, {Component} from 'react'
+import PropTypes from 'prop-types'
+import LeafletMap from './leaflet/map'
+import {mapViewModes} from './leaflet/constants'
+
+class MapBasic extends Component {
+  constructor (props) {
+    super(props)
+    this.setMapEventDefinition(props)
+    this.mapInstance = undefined
+    this.isHeatmapVisible = false
+    this.isSatelliteView = false
+  }
+
+  setMapEventDefinition () {
+    this.mapEventList = [
+      {
+        name: 'leaflet_map_click',
+        handleFunction: (evt) => this.props.onMapClick && this.props.onMapClick(evt.detail)
```

Comment:
In order to keep simple the component, I would put the `onMap...` method as a defaultProp with an empty function.

```
const NO_OP = () => {}
Map.defaultProps = {
  onMapClick: NO_OP
}
```

So you simplify your code like this as the method will exist always.
```
handleFunction: (evt) => this.props.onMapClick(evt.detail)
```

I even would like to see that:
```
handleFunction: this.props.onMapClick
```

And extract the `detail` from the `evt` on the prop instead. But I assume that you're doing this to keep compatibility.

Reply:
Totally agree with the first point! not so much on the second one.

As you said I've modified as much as possible without break anything on FC current map implementation. In the near future there are a bunch of things and contracts that should be simplified to dismiss big amounts of code to be executed.